src/common/config/index.js

```diff
@@ -0,0 +1,358 @@
+    delete this.combinedData.defaultTeam;
+    delete this.combinedData.defaultTeams;
+
+    // IMPORTANT: properly combine teams from all sources
+    const combinedTeams = [];
+
+    // - start by adding default teams from buildConfig, if any
+    if (this.buildConfigData.defaultTeams && this.buildConfigData.defaultTeams.length) {
+      combinedTeams.push(...this.buildConfigData.defaultTeams);
+    }
+
+    // - add GPO defined teams, if any
+    if (this.GPOConfigData.teams && this.GPOConfigData.teams.length) {
+      combinedTeams.push(...this.GPOConfigData.teams);
+    }
+
+    // - add locally defined teams only if server management is enabled
+    if (this.enableServerManagement) {
+      combinedTeams.push(...this.localConfigData.teams);
+    }
+
+    this.combinedData.teams = combinedTeams;
+    this.combinedData.localTeams = this.localConfigData.teams;
+    this.combinedData.buildTeams = this.buildConfigData.defaultTeams;
+    this.combinedData.GPOTeams = this.GPOConfigData.teams;
+  }
+
+  /**
+   * Returns the provided list of teams with duplicates filtered out
+   *
+   * @param {array} teams array of teams to check for duplicates
+   */
+  filterOutDuplicateTeams(teams) {
+    let newTeams = teams;
+    const uniqueURLs = new Set();
+    newTeams = newTeams.filter((team) => {
+      return uniqueURLs.has(team.url) ? false : uniqueURLs.add(team.url);
+    });
+    return newTeams;
+  }
+
+  /**
+   * Returns the provided array fo teams with existing teams filtered out
+   * @param {array} teams array of teams to check for already defined teams
+   */
+  filterOutPredefinedTeams(teams) {
+    let newTeams = teams;
+
+    // filter out predefined teams
+    newTeams = newTeams.filter((newTeam) => {
+      return this.predefinedTeams.findIndex((existingTeam) => newTeam.url === existingTeam.url) === -1; // eslint-disable-line max-nested-callbacks
+    });
+
+    return newTeams;
+  }
+
+  // helper functions
+
+  readFileSync(filePath) {
+    return JSON.parse(fs.readFileSync(filePath, 'utf8'));
+  }
+
+  writeFile(filePath, configData, callback) {
+    if (configData.version !== this.defaultConfigData.version) {
+      throw new Error('version ' + configData.version + ' is not equal to ' + this.defaultConfigData.version);
+    }
+    const json = JSON.stringify(configData, null, '  ');
+    fs.writeFile(filePath, json, 'utf8', callback);
+  }
+
+  writeFileSync(filePath, config) {
+    if (config.version !== this.defaultConfigData.version) {
+      throw new Error('version ' + config.version + ' is not equal to ' + this.defaultConfigData.version);
+    }
+
+    const dir = path.dirname(filePath);
+    if (!fs.existsSync(dir)) {
+      fs.mkdirSync(dir);
+    }
+
+    const json = JSON.stringify(config, null, '  ');
+    fs.writeFileSync(filePath, json, 'utf8');
+  }
+
+  merge(base, target) {
+    return Object.assign({}, base, target);
+  }
+
+  copy(data) {
+    return Object.assign({}, data);
+  }
+
+  deepCopy(data) {
```

Comment:
after removing the previous use of deepCopy is this still used anywhere?

Reply:
Probably not, kept it for legacy purposes. Will remove.